File: agritwin-ai/types/farm.ts
export interface Farm {
  id?: string | number;
  userId?: string | number;
  farmerName: string;
  farmName: string;
  location: string;
  size: number; // in acres
  soilType: string;
  waterSource: string;
  currentCrop: string;
  plantingDate: string; // ISO string or Date
  createdAt?: string;
  updatedAt?: string;
}

File: agritwin-ai/backend/app/schemas/farm.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class FarmBase(BaseModel):
    farm_name: str
    location: str
    farm_size: float
    soil_type: str
    water_source: str
    current_crop: str
    planting_date: datetime

class FarmCreate(FarmBase):
    pass

class FarmUpdate(BaseModel):
    farm_name: Optional[str] = None
    location: Optional[str] = None
    farm_size: Optional[float] = None
    soil_type: Optional[str] = None
    water_source: Optional[str] = None
    current_crop: Optional[str] = None
    planting_date: Optional[datetime] = None

class FarmResponse(FarmBase):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True
